src/domain/entities/PageBinding.ts

```diff
@@ -0,0 +1,117 @@
+import { generateUid } from "../../data/utils/uid";
+import { Maybe } from "../../types/utils";
+
+export const EVENT_TYPE = {
+    click: "click",
+    focus: "focus",
+    all: "all",
+} as const;
+
+const EVENT_TYPES = [EVENT_TYPE.click, EVENT_TYPE.focus, EVENT_TYPE.all] as const;
+const BINDING_TYPES = ["event", "section"] as const;
+
+export type EventType = typeof EVENT_TYPES[number];
+export type BindingType = typeof BINDING_TYPES[number];
+
+export const EventTypeModel: Codec<EventType> = Schema.oneOf([
+    Schema.exact("focus"),
+    Schema.exact("click"),
+    Schema.exact("all"),
+]);
+
+type BaseBinding = {
+    id: string;
+    description: Maybe<string>;
+};
+
+export type EventBinding = BaseBinding & {
+    type: "event";
+    trainingIdentifiers: string;
+    eventType: EventType;
+};
+
+export type SectionBinding = BaseBinding & {
+    type: "section";
+    urlPattern: string;
+};
+
+export type PageBinding = EventBinding | SectionBinding;
+
+const baseEventBinding = {
+    id: Schema.nonEmptyString,
+    description: Schema.optional(Schema.string),
+};
+
+export const EventBindingModel: Codec<EventBinding> = Schema.object({
+    ...baseEventBinding,
+    type: Schema.exact("event"),
+    trainingIdentifiers: Schema.string,
+    eventType: EventTypeModel,
+});
+
+export const SectionBindingModel: Codec<SectionBinding> = Schema.object({
+    ...baseEventBinding,
+    type: Schema.exact("section"),
+    urlPattern: Schema.nonEmptyString,
+});
+
+export const PageBindingModel: Codec<PageBinding> = Schema.oneOf([EventBindingModel, SectionBindingModel]);
+
+const defaultEventBinding = (): EventBinding => ({
+    id: generateUid(),
+    description: undefined,
+    type: "event",
+    trainingIdentifiers: "",
+    eventType: "click",
+});
+
+const defaultSectionBinding = (): SectionBinding => ({
+    id: generateUid(),
+    description: undefined,
+    type: "section",
+    urlPattern: "*/*",
+});
+
+export function getDefaultBinding(type: BindingType): PageBinding {
+    switch (type) {
+        case "section":
+            return defaultSectionBinding();
+        case "event":
+        default:
+            return defaultEventBinding();
+    }
+}
+
+export function urlPatternToRegex(pattern: string): RegExp {
+    const escaped = pattern.replace(/[.+?^${}()|[\]\\]/g, "\\$&");
+    const regexPattern = escaped
+        .replace(/\*\*/g, "<!GLOBSTAR!>")
+        .replace(/\*/g, "[^/]*")
+        .replace(/<!GLOBSTAR!>/g, ".*");
+
+    return new RegExp("^(?:/#)?" + regexPattern + "$");
+}
+
+export function matchesUrlPattern(currentPath: string, pattern: string): boolean {
+    const normalizedPath = currentPath.replace(/^\/#/, "").split("?")[0];
+    if (!normalizedPath) return false;
+
+    const regex = urlPatternToRegex(pattern);
+    return regex.test(normalizedPath);
+}
+
+export function isEventBinding(binding: PageBinding): binding is EventBinding {
+    return binding.type === "event";
```

Comment:
That’s very convenient for later usage with filter(isEventBinding). However, note that “is” assertion functions are beyond what TypeScript can fully type-check. For example, if you write return 2 > 1, TypeScript will happily accept it (but it will be wrong).